### ISSUE TITLE
Users: define requirements to replace Calypso "All Users/Users" with the standard WordPress "Users/All Users" view

### ISSUE THREAD

**Issue by annezazu** [Automattic/wp-calypso#95653]
> \[!IMPORTANT\]<br>**Project Thread: pdgR5j-18N-p2**

As part of the Untangling Calypso epic, this seeks to define requirements and any blockers for replacing [wordpress.com/people/team/view](<http://wordpress.com/people/team/view>) with the standard WordPress wp-admin/users.php  view for new users. This will reduce confusion navigating between Calypso and proper WordPress admin, offering a more expected experience.

Included in this should also be replacing Users > Add new user ([wordpress.com/people/new/](<http://wordpress.com/people/new/>)) with /wp-admin/user-new.php. Mentions of the differences there are noted below.

> \[!IMPORTANT\]<br>The outcome of this task is to identity the list of features that need to be migrated to the WP-Admin view in [WordPress.com](<http://WordPress.com>)

#### Add new user

* Ability to send a custom message in Calypso that's not available in Core when adding a new user at [wordpress.com/people/new/](<http://wordpress.com/people/new/>).
* Ability to mark new user as contractor/freelancer not available in Core
* Core requires both a username and email address when adding a user
* [WordPress.com](<http://WordPress.com>) Simple sites that are marked as private can also add ["Viewer" roles](<https://wordpress.com/support/invite-people/user-roles/#viewer>) at [http://calypso.localhost:3000/people/new/\[WP_SITE\]](<http://calypso.localhost:3000/people/new/%5BWP_SITE%5D>). Atomic sites use Core's user-new.php page and do not have this feature.

#### All users

* Ability to view a contractor in the list of users ([need to create an outside of organization contractor label](<https://wordpress.com/support/invite-people/user-roles/#contractor-freelancer-consultant-or-agency>)).
* Ability to see pending invites available in Calypso.
* Ability to complete bulk actions available in Core (delete, send password reset, change user role).

Please edit to add more! This is just a start.

**Comment by Copons** [comment]
Mmh, when I looked at this page, I was tempted to just drop the Calypso version, but @annezazu, you make good points. I don't think I'm familiar enough with the area to make an educated decision.

It seems to me that the main differences concern "external" users, who are typically a bit neglected on vanilla Core. For example, Subscribers (external consumers) and Contractors (external authors).

I wonder if @simison has more insights on this, with his newsletters expertise. 🙂 

I'm also not entirely sure what Contractors do. It was [introduced a long time ago](https://github.com/Automattic/wp-calypso/pull/32805), and the original owners left A8C. Maybe @stephanethomas might have some additional context?

**Comment by Copons** [comment]
> Subscriber management is still in Calypso+Jetpack Cloud, but soon moving to WP Admin using DataViews:

Love it!

@simison in the context of newsletters, do you use any Calypso functionalities like "send a custom message when adding a new user" or "see pending invites"?

**Comment by simison** [comment]
Nope, separate APIs too.

**Comment by annezazu** [comment]
Incredible news! I couldn't quite figure out what was going on there so thank you for clarifying.

**Comment by Copons** [comment]
Connecting with another issue: https://github.com/Automattic/dotcom-forge/issues/7934

**Comment by ramonjd** [comment]
What would have to be turned off in WPCOM/Jetpack so that the default `user-new.php` and `user.php` are used? I'm seeing [this Jetpack file](https://github.com/Automattic/jetpack/blob/46e21fba9e02154b8362b1f7edb75171dc01b21e/projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/class-wpcom-admin-bar.php#L33) that references `user-new.php` for example, but there a quite a few in the WPCOM backend.

I could see Core adding an optional email message - seems like a good addition.

As far as I can tell from [WordPress Core code](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/user-new.php#L440), "inviting" users is a multisite-only thing, and there doesn't seem to be a "pending" flag in the [table](https://github.com/WordPress/wordpress-develop/blob/4e1752dee97eb3454c7a6da9176b8890d7c371a8/src/wp-admin/includes/class-wp-users-list-table.php#L17).

**Comment by tellthemachines** [comment]
In addition to what Anne listed in the description, in Core adding a new user requires _both_ a username and email address. In dotcom you add _either_ email or username, if you're adding an existing dotcom user. If it's not an existing user, you add the email and the person gets invited to join dotcom.

For multisite instances in Core, there's the option to add either an existing user from the network or a new user, but it's still required to input a username for that new user:

<img width="1112" alt="Image" src="https://github.com/user-attachments/assets/1fab2bec-147c-42a1-9023-07ef5c18c8bc" />

We could possibly leverage the `pre_user_login` filter to auto-generate the new user name as we seem to do in dotcom, but it would be a pretty bad experience to force our users to fill in that field only to then ignore it, so I'm not sure we can use the Core page 100% as is.

**Comment by ramonjd** [comment]
Another thing I noticed:

WordPress.com Simple sites that are marked as private can also add ["Viewer" roles](https://wordpress.com/support/invite-people/user-roles/#viewer) at `http://calypso.localhost:3000/people/new/[WP_SITE]`

<img width="384" alt="Image" src="https://github.com/user-attachments/assets/f0ad9cab-a05d-4dc8-85e1-a21bea7650d7" />

Atomic sites use Core's `user-new.php` page and do not have this feature.

In Core the user role select box is rendered using the output of [wp_dropdown_roles](https://github.com/WordPress/wordpress-develop/blob/86f31c81668c0a680b6db275b41298f7e8513389/src/wp-admin/user-new.php#L640).

The [function wp_dropdown_roles](https://github.com/WordPress/wordpress-develop/blob/86f31c81668c0a680b6db275b41298f7e8513389/src/wp-admin/includes/template.php#L976), calls [get_editable_roles](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/user.php#L278). So I guess Simple sites could hook into https://developer.wordpress.org/reference/hooks/editable_roles/ and add the viewer role.

**Comment by aaronrobertshaw** [comment]
> WordPress.com Simple sites that are marked as private can also add ["Viewer" roles](https://wordpress.com/support/invite-people/user-roles/#viewer)

Also, if a private WordPress.com Simple site has "Viewer" users added, they don't show on the wp-admin All User page.

| WP Admin | Calypso |
|---|---|
| <img alt="Image" src="https://github.com/user-attachments/assets/5046202f-03d2-4ba0-a174-72265b45bb8d" /> | <img alt="Image" src="https://github.com/user-attachments/assets/8efbab4a-06e5-4afc-a442-c543c1b13824" /> |

**Comment by ramonjd** [comment]
> Clicking on the Viewer tab shows an empty screen despite that new user showing with the Viewer role under Team.

Thanks for investigating! I've been looking at some of the [current bugs](https://github.com/Automattic/wp-calypso/issues?q=sort%3Aupdated-desc%20is%3Aissue%20is%3Aopen%20label%3A%22%5BFeature%5D%20User%20Management%22&page=1), and I'm hoping defaulting to the wp-admin screen might "fix" a lot them by default, e.g., ["bulk delete"](https://github.com/Automattic/wp-calypso/issues/76745) 

Well, I guess there'll be different bugs 😄 

Looks there are some other WPCOM roles too: [email subscribers and followers](https://github.com/Automattic/wp-calypso/issues/71508). I can't seem to find them using `get_editable_roles()`. Followers look to be stored in post_meta via the ActivityPub mu plugin - this could be also tricky to port over 🤔

**Comment by ramonjd** [comment]
Feature comparison over at pdgR5j-199-p2

The outcome will be an MVP. Suggested features to include:

- Subscribers to live permanently under the Jetpack menu
- integrate Viewers into the users table. First path is to get the users to display in the table. See 171239-ghe-Automattic/wpcom and https://github.com/Automattic/jetpack/pull/41447
- Align the add users form (add viewers, custom message, add by username, contractor status (?))
- Decide where and how to show pending invites
- Investigate why viewers are not being copied during Atomic transfer (might be good for another team with more experience to do this in parallel)
- Align profile page

**Comment by Copons** [comment]
### 2025-03-03

The project is paused pending review of the Untangling focus.

**Comment by ramonjd** [comment]
closing pending further project updates.
